game-app/game-core/src/main/java/games/strategy/triplea/ui/TechnologyPanel.java

```diff
@@ -0,0 +1,288 @@
+        if (gameDataSync.getResourceList().getResource(Constants.TECH_TOKENS) != null) {
+          useTech = true;
+          dataTable = new String[numTechs + 1][colList.length + 2];
+        } else {
+          dataTable = new String[numTechs][colList.length + 1];
+        }
+      }
+      /* Load the technology -> row mapping */
+      int row = 0;
+      if (useTech) {
+        rowMap.put("Tokens", row);
+        dataTable[row][0] = "Tokens";
+        row++;
+      }
+      final List<TechAdvance> techAdvances =
+          TechAdvance.getTechAdvances(gameDataSync.getTechnologyFrontier(), null);
+      for (final TechAdvance tech : techAdvances) {
+        rowMap.put(tech.getName(), row);
+        dataTable[row][0] = tech.getName();
+        row++;
+      }
+      return dataTable;
+    }
+
+    private void clearAdvances() {
+      /* Initialize the table with the tech names */
+      for (int i = 0; i < data.length; i++) {
+        for (int j = 1; j <= colList.length; j++) {
+          data[i][j] = "";
+        }
+      }
+    }
+
+    private void initColList() {
+      final List<GamePlayer> players = gameData.getPlayerList().getPlayers();
+      colList = new String[players.size()];
+      for (int i = 0; i < players.size(); i++) {
+        colList[i] = players.get(i).getName();
+      }
+      Arrays.sort(colList);
+    }
+
+    synchronized void update() {
+      clearAdvances();
+      // copy so that the object doesn't change underneath us
+      final GameData gameDataSync = TechnologyPanel.this.gameData;
+      try (GameData.Unlocker ignored = gameDataSync.acquireReadLock()) {
+        for (final GamePlayer pid : gameDataSync.getPlayerList().getPlayers()) {
+          if (colMap.get(pid.getName()) == null) {
+            throw new IllegalStateException(
+                "Unexpected player in GameData.getPlayerList()" + pid.getName());
+          }
+          final int col = colMap.get(pid.getName());
+          int row = 0;
+          if (gameDataSync.getResourceList().getResource(Constants.TECH_TOKENS) != null) {
+            final int tokens = pid.getResources().getQuantity(Constants.TECH_TOKENS);
+            data[row][col] = Integer.toString(tokens);
+          }
+          final List<TechAdvance> advancesAll =
+              TechAdvance.getTechAdvances(gameDataSync.getTechnologyFrontier());
+          final List<TechAdvance> has =
+              TechAdvance.getTechAdvances(gameDataSync.getTechnologyFrontier(), pid);
+          for (final TechAdvance advance : advancesAll) {
+            if (!has.contains(advance)) {
+              row = rowMap.get(advance.getName());
+              data[row][col] = "-";
+            }
+          }
+          for (final TechAdvance advance :
+              TechTracker.getCurrentTechAdvances(pid, gameDataSync.getTechnologyFrontier())) {
+            row = rowMap.get(advance.getName());
+            data[row][col] = "X";
```

Comment:
Define constant like `SIGN_TECH_ENABLED` and annotate as NonNls (not language dependent).